src/main/java/io/kubernetes/KabaneroClient.java

```diff
@@ -95,10 +100,43 @@ private static ApiClient getApiClient() throws IOException, GeneralSecurityExcep
         ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS).allEnabledCipherSuites().build();
         client.getHttpClient().setConnectionSpecs(Collections.singletonList((spec)));
 
+        client.getHttpClient().setReadTimeout(0, TimeUnit.SECONDS); // infinite timeout
+
         Configuration.setDefaultApiClient(client);
         return client;
     }
 
+    /**
+     * Using the Watch API to watch for changes in the kabanero namespace.
+     * 
+     * @return whether or not changes were made in the namespace
+     */
+    private static boolean refreshInstance() throws IOException, GeneralSecurityException, ApiException {
```

Comment:
you shouldn't need a refreshInstance like this, you would want a watcher on each piece of data (instead of the whole object)